core/chain/abci.go

```diff
@@ -113,19 +140,18 @@ func (app *CoreApplication) FinalizeBlock(ctx context.Context, req *abcitypes.Fi
 				txs[i] = &abcitypes.ExecTxResult{Code: 2}
 			}
 			txs[i] = &abcitypes.ExecTxResult{Code: abcitypes.CodeTypeOK}
-			continue
-		}
-
-		if code := app.isValid(tx); code != 0 {
-			logger.Errorf("Error: invalid transaction index %v", i)
-			txs[i] = &abcitypes.ExecTxResult{Code: code}
-		} else {
+		} else if app.isRollupTx(tx) {
+			if err := app.indexRollupTx(ctx, tx); err != nil {
+				logger.Error("Failed to index rollup", "error", err)
+				txs[i] = &abcitypes.ExecTxResult{Code: 1}
+			} else {
+				txs[i] = &abcitypes.ExecTxResult{Code: abcitypes.CodeTypeOK}
+			}
```

Comment:
instead of adding another if here, since this is a protobuf message let's pass it into finalize event and add another case here
https://github.com/AudiusProject/audius-protocol/blob/main/core/chain/finalize.go#L10-L17

```
	case *gen_proto.Event_Plays:
        case *gen_proto.SlaRollup:
                   //... impl
        case *gen_proto.SlaNodeReport:
                   //... impl
	case *gen_proto.Event_RegisterNode:
		return core.finalizeRegisterNode(ctx, msg)
	}
```